### ISSUE TITLE
How to modify to use in RetinaNet?

### ISSUE THREAD

**Issue by chandlerbing65nm** [RuoyuChen10/objectdetection-saliency-maps#1]
How can I modify the [gradcam_yolov3.py](https://github.com/RuoyuChen10/objectdetection-saliency-maps/blob/main/tutorial/gradcam-yolov3.ipynb) to use in RetinaNet?

**Comment by chandlerbing65nm** [comment]
When I tried to modify the config file, using RetinaNet config, an error occurs:

![image](https://user-images.githubusercontent.com/62779617/158602059-54280f1e-2d00-41bc-9dc2-e9d9154ef444.png)

**Comment by chandlerbing65nm** [comment]
@RuoyuChen10 do you have the working demo of gradcam for faster-RCNN?

**Comment by chandlerbing65nm** [comment]
> > @RuoyuChen10 do you have the working demo of gradcam for faster-RCNN?
> 
> Yes, I will submit the `.ipynb` demo before Beijing time 3/18/2022 24:00.

Thank you

**Comment by RuoyuChen10** [comment]
> > > @RuoyuChen10 do you have the working demo of gradcam for faster-RCNN?
> > 
> > 
> > Yes, I will submit the `.ipynb` demo before Beijing time 3/18/2022 24:00.
> 
> Thank you

Or you can find an implementation here: [https://github.com/yizt/Grad-CAM.pytorch](https://github.com/yizt/Grad-CAM.pytorch), there is both faster R-CNN and retinanet in the project based on the `detectron2` framework. I may submit `mmdetection` later due to the ddl of ACM MM. Maybe you can offer an email to me and I send you the `ipynb` file without readme.

**Comment by chandlerbing65nm** [comment]
> 

Yes, I viewed that implementation in `detectron2` and thought to re-implement it in mmdetection. Thankfully, I don't need to do anymore since you've done it already. 

Thank you for sharing, my email is `chandlertimmdoloriel@gmail.com`

**Comment by Drew-bw** [comment]
Hi! I have viewed the implementation in detectron2 but failed to re-implement it in mmdetection. Could you provide me with the .ipynb demo on retinanet?  my email is baowei97@163.com. Thank you!

**Comment by RuoyuChen10** [comment]
> Hi! I have viewed the implementation in detectron2 but failed to re-implement it in mmdetection. Could you provide me with the .ipynb demo on retinanet? my email is [baowei97@163.com](mailto:baowei97@163.com). Thank you!

Sorry, I currently don't have `retinanet-gradcam` implementation on `mmdetection`. I will submit it to this repo if I have time to implement it.

**Comment by Drew-bw** [comment]
Thank you all the same！



---- 回复的原邮件 ----
| 发件人 | Ruoyu ***@***.***> |
| 日期 | 2022年04月17日 16:33 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.******@***.***> |
| 主题 | Re: [RuoyuChen10/objectdetection-saliency-maps] How to modify to use in RetinaNet? (Issue #1) |

Hi! I have viewed the implementation in detectron2 but failed to re-implement it in mmdetection. Could you provide me with the .ipynb demo on retinanet? my email is ***@***.*** Thank you!

Sorry, I currently don't have retinanet-gradcam implementation on mmdetection. I will submit it to this repo if I have time to implement it.

**Comment by RuoyuChen10** [comment]
> > 
> 
> @RuoyuChen10 Yes, I viewed that implementation in `detectron2` and thought to re-implement it in mmdetection. Thankfully, I don't need to do anymore since you've done it already.
> 
> Thank you for sharing, my email is `chandlertimmdoloriel@gmail.com`

I have updated the Grad-CAM for CUDA computing. See `interpretation` dir.

**Comment by RuoyuChen10** [comment]
I have finised the demo based on faster R-CNN.

**Comment by RuoyuChen10** [comment]
> How can I modify the [gradcam_yolov3.py](https://github.com/RuoyuChen10/objectdetection-saliency-maps/blob/main/tutorial/gradcam-yolov3.ipynb) to use in RetinaNet?

I have released an example for retinanet, have a look.
